### ISSUE TITLE
CircleCI Temp Workaround

### ISSUE THREAD

**Issue by typerror** [erikdoe/ccmenu#102]
Hello everyone! 
CircleCI support page in the [wiki page](https://github.com/erikdoe/ccmenu/wiki/Supported-Servers) is not updated (404).
So I'm giving a temporary explanation for those struggling with setting up circle ci projects.

- go to circle ci api webpage and create a token
 
  - https://circleci.com/gh/user/project/edit#api 

    - replace "user" with your username 

    - replace "project" with your project name

- open ccmenu > prerences > projects tab > +

  - open ccmenu, go to preferences, projects tab and add a project

- In the Feed URL insert

  - https://circleci.com/gh/user/project/tree/branch.cc.xml?circle-token=your_token

    - replace "user" with your username

    - replace "project" with your project name

    - replace "branch" with the branch you wish to monitor
    
      - example master.cc.xml

    - replace "your_token" with the previously generated token

    - check "use URL as entered above" 

<img width="427" alt="ccmenucircleci" src="https://user-images.githubusercontent.com/32941768/46918862-ca139780-cfd7-11e8-9e8e-622f9f761e91.png">

**Comment by jhirn** [comment]
My Circle project stopped updating recently for me and a teammate. I tried replacing the token with a fresh one but no matter what it just says my project is sleeping. Not sure if Circle is in the middle up updating something (thus the 404) but I'm up to date with the description here and still not getting any status updates.

**Comment by erikdoe** [comment]
Is the JSON response structured in the same way as the old XML-based format or do they use new field names and values, too? If you have access to a CircleCI build, I'd appreciate it if you could post a sample response here.

**Comment by hajder** [comment]
@erikdoe oh wow, thanks for a quick response. Here is, slightly sanitized, response

```
[ {
  "compare" : "https://github.com/user/project/compare/ee6dd61f7792...c90b11921995",
  "previous_successful_build" : {
    "build_num" : 1607,
    "status" : "success",
    "build_time_millis" : 410462
  },
  "build_parameters" : null,
  "oss" : false,
  "committer_date" : "2018-12-20T09:05:47+01:00",
  "body" : "",
  "usage_queued_at" : "2018-12-20T08:14:47.907Z",
  "fail_reason" : null,
  "retry_of" : 1607,
  "reponame" : "reponame",
  "ssh_users" : [ ],
  "build_url" : "https://circleci.com/gh/user/project/1609",
  "parallel" : 1,
  "failed" : false,
  "branch" : "master",
  "username" : "user",
  "author_date" : "2018-12-20T09:05:47+01:00",
  "why" : "retry",
  "user" : {
    "is_user" : true,
    "login" : "username",
    "avatar_url" : "https://avatars3.githubusercontent.com/u/12345?v=4",
    "name" : "First name Last name",
    "vcs_type" : "github",
    "id" : 12345
  },
  "vcs_revision" : "c90b11921995906478fe6dea00fe21d41e854a09",
  "vcs_tag" : null,
  "build_num" : 1609,
  "infrastructure_fail" : false,
  "committer_email" : "user@example.com",
  "has_artifacts" : true,
  "previous" : {
    "build_num" : 1607,
    "status" : "success",
    "build_time_millis" : 410462
  },
  "status" : "success",
  "committer_name" : "First name Last name",
  "retries" : null,
  "subject" : "commit msg",
  "vcs_type" : "github",
  "timedout" : false,
  "dont_build" : null,
  "lifecycle" : "finished",
  "no_dependency_cache" : false,
  "stop_time" : "2018-12-20T08:20:24.164Z",
  "ssh_disabled" : true,
  "build_time_millis" : 334576,
  "picard" : {
    "build_agent" : {
      "image" : "circleci/picard:0.1.1250-22bf9f5d",
      "properties" : {
        "build_agent" : "0.1.1250-22bf9f5d",
        "executor" : "docker"
      }
    },
    "resource_class" : {
      "cpu" : 2.0,
      "ram" : 4096,
      "class" : "medium"
    },
    "executor" : "docker"
  },
  "circle_yml" : {
    "string" : "whole lotta secret stuff in json format"
  },
  "messages" : [ ],
  "is_first_green_build" : false,
  "job_name" : null,
  "start_time" : "2018-12-20T08:14:49.588Z",
  "canceler" : null,
  "all_commit_details" : [ {
    "committer_date" : "2018-12-20T00:42:16+01:00",
    "body" : "",
    "branch" : "master",
    "author_date" : "2018-12-20T00:42:16+01:00",
    "committer_email" : "user@example.com",
    "commit" : "6c832597973786dcb15f72614b22284247be5217",
    "committer_login" : "username",
    "committer_name" : "First name Last name",
    "subject" : "deploy from master",
    "commit_url" : "https://github.com/user/project/commit/6c832597973786dcb15f72614b22284247be5217",
    "author_login" : "username",
    "author_name" : "First name Last name",
    "author_email" : "user@example.com"
  }, {
    "committer_date" : "2018-12-20T09:05:47+01:00",
    "body" : "",
    "branch" : "master",
    "author_date" : "2018-12-20T09:05:47+01:00",
    "committer_email" : "user@example.com",
    "commit" : "c90b11921995906478fe6dea00fe21d41e854a09",
    "committer_login" : "username",
    "committer_name" : "First name Last name",
    "subject" : "commit msg",
    "commit_url" : "https://github.com/user/project/commit/c90b11921995906478fe6dea00fe21d41e854a09",
    "author_login" : "username",
    "author_name" : "First name Last name",
    "author_email" : "user@example.com"
  } ],
  "platform" : "2.0",
  "outcome" : "success",
  "vcs_url" : "https://github.com/user/project",
  "author_name" : "First name Last name",
  "node" : null,
  "queued_at" : "2018-12-20T08:14:47.936Z",
  "canceled" : false,
  "author_email" : "user@example.com"
}]
```

**Comment by hajder** [comment]
So for the attributes needed for ccmenu

name: `build_num`?
activity: `status`
lastBuildStatus: `previous.status`
lastBuildLabel: `previous.build_num`?
lastBuildTime: not available
nextBuildTime: not available?
webUrl: `build_url`

**Comment by phoet** [comment]
@erikdoe is this going to be a christmas gift? 😸

**Comment by phoet** [comment]
@kornysietsma true, that works with your personal api-token. i used to do that the same way.

**Comment by johnnylai** [comment]
@kornysietsma thx for the tip. It works for me. But this seems to only work with the default branch for a repo. It seems not able for user to select which branch they want to watch. 

Still, thanks for the info.

**Comment by erikdoe** [comment]
We're in the process of migrating these wiki pages to a website on the cctray format: https://github.com/build-canaries/cctray

It would be fantastic if someone could write up the current approach to making CCMenu work with CircleCI and add this via a pull request over there.

**Comment by sikachu** [comment]
I'm glad I found this issue! I really didn't know that personal `/cc.xml` exists!

I opened https://github.com/build-canaries/cctray/pull/10 which updates the documentation regarding how to setup CCMenu/CCTray to use this instead.

Thank you again for this amazing project!

**Comment by ospfranco** [comment]
Hey, I've been using:

https://circleci.com/cc.xml?circle-token=my_api_token

For quite some time, but it no longer is enough for my workflow, I need to monitor more than one branch in the project, is there any other possible way to make per branch configurations work?

https://circleci.com/gh/user/project/tree/branch.cc.xml?circle-token=your_token

**Comment by hadees** [comment]
So I think I figured out how to do this.  If you look at [CircleCI docs](https://github.com/circleci/circleci-docs/blob/master/jekyll/_cci1/polling-project-status.md#configuration) you can actually craft the url with `cc.xml` anyway you want. So in my case `https://circleci.com/gh/foo/bar/tree/master.cc.xml?circle-token=foobar`got me `master`.

**Comment by hubertjakubiak** [comment]
@ospfranco Any update in 2021 how to dynamically monitor my branches? Did you find another way to achieve that?

**Comment by ospfranco** [comment]
I have, I created my own app, it's called CI Demon:

https://apps.apple.com/de/app/ci-demon/id1560355863?mt=12

It's a lot more complete than CCMenu and obviously updated (as of 2021), not only it can show per branch configuration, you can also filter via regexes, shows additional info, can trigger rebuilds, a lot's of other goodies.

With CCMenu, nope, there is no other way.

**Comment by rodrigodealer** [comment]
Does anyone know if these APIs are still working for CircleCI? 

I've been trying this, but I'm getting a 401 with "Invalid token provided.", even though I'm sending the basic auth params and the personal token as well.
